### PR TITLE
feat: add analysis modal

### PR DESCRIPTION
**Pull request by emily-blixt-ck**
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.

* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  

* [X] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)

* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.

* [ ] My builds are green. Try syncing with master if they are not. 

* [X] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

Screenshots:

Summary panel
Screenshot 2023-11-16 at 4 26 44 PM

Metric chart demonstrating successes and errors
Screenshot 2023-11-16 at 2 48 15 PM

Table view demonstrating successes and errors
Screenshot 2023-11-16 at 2 48 23 PM

Successful metric chart
Screenshot 2023-11-16 at 2 49 25 PM

Failed metric table
Screenshot 2023-11-16 at 2 52 17 PM

### CODE REVIEW

**Comment by ashutosh16** [comment]
@emily-blixt-ck  Analysis tab need to be color code based on the health status. For e.g.
Degraded - Red
Healthy - Green
Running - Blue
Inconclusive - yellow
Screenshot 2023-12-05 at 9 46 22 AM

File: ui/src/app/components/analysis-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
```
Comment (ashutosh16) [inline ui/src/app/components/analysis-modal/index.tsx:1]: Would it better to rename the index.tsx to analysis-modal.tsx

Comment (emily-blixt-ck) [reply ui/src/app/components/analysis-modal/index.tsx]: Yes, that makes sense for consistency. I renamed it.

**Comment by emily-blixt-ck** [comment]
> @emily-blixt-ck Analysis tab needs to be color code based on the health status. For e.g. Degraded - Red Healthy - Green Running - Blue Inconclusive - yellow Screenshot 2023-12-05 at 9 46 22 AM

Thanks for pointing this out! Added a simple status indicator (this doesn't include a sub-status dot, but we could add that too if we want).

Screenshot 2023-12-05 at 1 08 32 PM

File: ui/src/app/components/analysis-modal/panels/index.tsx [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export {default as AnalysisPanel} from './analysis-panel';
```
[the remark this answers is not in the document]
Comment (emily-blixt-ck) [reply ui/src/app/components/analysis-modal/panels/index.tsx:1]: Yes. This is the Summary Panel that you see when you open the modal. Now that I've typed that... I could rename it to SummaryPanel if that would be more clear.

[the remark this answers is not in the document]
Comment (ashutosh16) [reply ui/src/app/components/analysis-modal/panels/index.tsx:1]: Yes, renaming would be better

[the remark this answers is not in the document]
Comment (emily-blixt-ck) [reply ui/src/app/components/analysis-modal/panels/index.tsx]: Done.

**Comment by ashutosh16** [comment]
@zachaller  Would you be able to run this locally and verify the UI?

**Review by zachaller — APPROVED** [review]

**Comment by danil-smirnov** [comment]
We are waiting for this to be released folks!

**Comment by zachaller** [comment]
[These](https://github.com/argoproj/argo-rollouts/pulls?q=is%3Aopen+is%3Apr+milestone%3Av1.7) are the last features that need to make it in for 1.7 then I should be good to cut a 1.7 release, I will try to get to them soon.

**Pull request merged**